src/Symfony/Component/Translation/DataCollector/TranslationDataCollector.php

```diff
@@ -88,6 +91,26 @@ public function getCountDefines()
     }
 
     /**
+     * Gets the translator locale.
+     *
+     * @return string
+     */
+    public function getLocale()
+    {
+        return !empty($this->data['locale']) ? $this->data['locale'] : '-';
```

Comment:
A quick comment about using `-` as a return value: the PHP methods of data collectors can be called from other PHP code (e.g. when [using the profiler in tests](https://symfony.com/doc/current/testing/profiling.html)). So I always prefer to return values optimized for PHP. In this case, `null` or an empty string (as you originally did) would be more convenient than returning `-`. The `-` symbol is just a visual detail used to let the user know that this value is empty ... so maybe is better to move that to the Twig template. Thanks!

(same comment for `getFallbackLocales()` ... retuning an empty array may be better)